lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -71,6 +71,9 @@ export class ClientCredentialClient extends BaseClient {
     private async getCachedAuthenticationResult(
         request: CommonClientCredentialRequest
     ): Promise<AuthenticationResult | null> {
+        // read the user-supplied cache into memory, if applicable
+        await this.cacheManager.readUserDefinedCacheIntoMemory(this.config.serializableCache, this.config.persistencePlugin);
```

Comment:
I think it's ok to invoke `beforeCacheAccess` / `afterCacheAccess` here.

Reply:
That's how I had it originally before I moved it to the cache manager. I'll move it back.

Reply:
I think short-term it's ok to put it here as it's consistent with other code. But it seems more appropriate to put this somewhere in more internal cache classes, that's why CacheManager seemed better. Imo ClientCredentialClient and ResponseHandler shouldn't need to know the proper steps to call / cache; these event handlers are internal implementation that should be abstracted away somwhere.

Reply:
CacheManager is intended as a generic abstraction for any cache lookups/storage/removal and the exact implementation is delegated to the platform layer (node in this case). Serialization and snapshots of cache are a part of node and not the common layer. 

I would prefer adding this code (if it is ever moved) to the `msal-node` cache layer and not in common.